docs/groovy-script/mods/mwc/crafting_station.md

```diff
@@ -0,0 +1,209 @@
+    setGroupGun()
+    setGroupBullet()
+    setGroupGrenade()
+    setGroupMagazine()
+    setGroupNormalAttachment()
+    setGroupModificationAttachment()
+    setGroup(CraftingGroup)
+    ```
+
+- `List<CraftingEntry>`. Input entries of the recipe, `double` is refund yield of the ingredient. Requires greater than 0.
+
+    ```groovy:no-line-numbers
+    input(IIngredient)
+    input(IIngredient...)
+    input(double, IIngredient)
+    input(double, IIngredient...)
+    input(Collection<IIngredient>)
+    ```
+
+- First validates the builder, returning `null` and outputting errors to the log file if the validation failed, then registers the builder and returns the registered object. (returns `null` or `com.paneedah.mwc.groovyscript.recipes.GSCraftingRecipe`).
+
+    ```groovy:no-line-numbers
+    register()
+    ```
+
+::::::::: details Example {open id="example"}
+```groovy:no-line-numbers
+mods.mwc.crafting_station.recipeBuilder()
+    .input(1, item('minecraft:clay'))
+    .output(item('minecraft:diamond'))
+    .register()
+
+mods.mwc.crafting_station.recipeBuilder()
+    .input(0.5, item('minecraft:gold_ingot'))
+    .output(item('minecraft:clay') * 2)
+    .register()
+```
+
+:::::::::
+
+::::::::::
+
+## Removing Recipes
+
+- Remove all recipes of Bullet Category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeAllBullet()
+    ```
+
+- Remove all recipes of Grenade Category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeAllGrenade()
+    ```
+
+- Remove all recipes of Gun Category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeAllGun()
+    ```
+
+- Remove all recipes of Magazine Category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeAllMagazine()
+    ```
+
+- Remove all recipes of Modification Attachment Category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeAllModificationAttachment()
+    ```
+
+- Remove all recipes of Normal Attachment Category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeAllNormalAttachment()
+    ```
+
+- Removes recipe that outputs matching item of given `ingredient` from bullet category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeBulletByOutput(IIngredient)
+    ```
+
+- Removes recipe that outputs matching item of given `ingredient` from grenade category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeGrenadeByOutput(IIngredient)
+    ```
+
+- Removes recipe that outputs matching item of given `ingredient` from gun category:
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeGunByOutput(IIngredient)
+    ```
+
+- Removes Recipe that outputs given `ingredient` from given category. valid categories = "GUN", "ATTACHMENT_NORMAL", "ATTACHMENT_MODIFICATION", "BULLET", "MAGAZINE", "GRENADE":
+
+    ```groovy:no-line-numbers
+    mods.mwc.crafting_station.removeInGroupByOutput(IIngredient, CraftingGroup)
```

Comment:
probably want a `removeInGroupByOutput(IIngredient, String)` method to go along with this.